docs/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst

```diff
@@ -148,12 +197,54 @@ Additional Notes
     Should you need to modify your connection string, you can manually edit the ``great_expectations/uncommitted/config_variables.yml`` file.
 
 #.
-    You can edit the  ``great_expectations/uncommitted/config_variables.yml``file to accomplish the connection configuration without using the CLI.  The entry would have the following format:
+    You can edit the  ``great_expectations/uncommitted/config_variables.yml`` file to accomplish the connection configuration without using the CLI.  The entry would have the following format:
+
+    **For "User and password authentication":**
+
+        .. code-block:: yaml
+
+            my_snowflake_db:
+                url: "snowflake://<user_login_name>:<password>@<account_name>/<database_name>/<schema_name>?warehouse=<warehouse_name>&role=<role_name>"
+
+    **For "Single sign-on authentication":**
+
+        .. code-block:: yaml
+
+            my_snowflake_db:
+                url: "snowflake://<myuser%40mydomain.com>:<password>@<account_name>/<database_name>/<schema_name>?authenticator=<externalbrowser or valid URL encoded okta url>&warehouse=<warehouse_name>&role=<role_name>"
+
+    **For "Key pair authentication":**
+
+        Snowflake requires the private key to be passed as "unencrypted rsa private key in DER format as bytes object",
+        this makes it hard to create a URL manually.
```

Comment:
I think we should just omit this sentence, since in general we prefer the more expanded format for storing credentials etc.

Reply:
done!